docs/allocation.md

```diff
@@ -0,0 +1,152 @@
+
+<i class="fa fa-pencil-square-o" aria-hidden="true"></i> **Note:** On operating systems that support paging, the VM allocates a single contiguous area that matches the value of `-Xms`. Additional memory is committed as the heap expands by using the paging process.
+
+Expansion occurs to maintain free space on the Java heap for object allocation. By default, the heap is expanded to maintain 30% free space, but this proportion can be adjusted by setting one of the following command-line options:
+
+- `-Xminf` determines the minimum proportion of the heap that must be free after garbage is removed.
+- `-Xmaxf` determines the maximum proportion of the heap that must be free after garbage is removed.
+
+If expansion is required, the amount of memory that the heap can expand by is controlled by the following command-line option:
+
+- `-Xmine` sets the minimum amount that the heap can expand by.
+- `-Xmaxe` sets the maximum amount that the heap can expand by. The default is unlimited expansion up to the maximum heap size (`-Xmx`).
+
+Expansion can also be triggered if more time is being spent on GC processing than is specified by the `-Xmaxt` option. In this case, the heap expands by an amount that provides 17% more free space, within the limits imposed by the `-Xmine` and `-Xmaxe` values.
+
+Heap shrinkage occurs under certain conditions and might be preceded by heap compaction. If the last three GC cycles caused a heap expansion, shrinkage does not occur. Otherwise, shrinkage is triggered when the proportion of free heap space that is specified by the `-Xmaxf` option is reached. The amount of memory to reduce the heap by is calculated to the nearest 1024-byte boundary, down to the minimum size specified for the initial Java heap (`-Xms`). To prevent heap shrinkage, set the `-Xmaxf` value to `1`, which sets the maximum free space allowed on the heap to 100%.
+
+When the heap shrinks, physical memory is not released unless paging is supported by the underlying operating system.
+
+
+### Compressed references
+
+On 64-bit systems, the VM can use compressed references to decrease the size of Java objects and make better use of the available space in the Java heap. By storing objects in a 32-bit representation, the object size is identical to a 32-bit object, which occupies a smaller memory footprint. These 4 byte (32-bit) compressed references are converted to 64-bit values at runtime with minimal overhead. Smaller objects enable larger heap sizes that result in less frequent garbage collection and improve memory cache utilization. Overall, the performance of 64-bit applications that store compressed rather than uncompressed 64-bit object references is significantly improved.
```

Comment:
Wording does not seem right here:
'By storing objects in a 32-bit representation, the object size is identical to a 32-bit object, which occupies a smaller memory footprint.'

Should be (i'm maybe being a bit too verbose, though):

By storing _references_ in a 32-bit representation, the total object size is identical to that in a 32-bit VM, which occupies a smaller memory footprint than that in 64-bit VM.

Also I don't know if it's in nature of English to say 'occupies footprint'. It's more natural to me 'creates footprint', or 'occupies memory' (without word footprint).

Reply:
Adapted to "By storing objects in a 32-bit representation, the object size is identical to that in a 32-bit VM, which creates a smaller memory footprint."